File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/TeleportLandingStatusChanged.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

/**
 * TeleportLandingStatusChanged
 * sim->dataserver
 * Sent from the region to the data server
 * to note that the region's teleportation landing status has changed
 *
 * <p>Template: {@code TeleportLandingStatusChanged Low 147 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class TeleportLandingStatusChanged extends SLMessage {
    public RegionData RegionData_Field;

    /** Block RegionData, Single. */
    public static class RegionData {
        public long RegionHandle; // U64
    }

    public TeleportLandingStatusChanged() {
        this.zeroCoded = false;
        this.RegionData_Field = new RegionData();
    }

    @Override
    public int CalcPayloadSize() {
        return 12;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleTeleportLandingStatusChanged(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 147 (TeleportLandingStatusChanged).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x00);
        byteBuffer.put((byte) 0x93);
        packLong(byteBuffer, this.RegionData_Field.RegionHandle);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.RegionData_Field.RegionHandle = unpackLong(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/TeleportLandingStatusChanged.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer

/**
 * TeleportLandingStatusChanged
 * sim->dataserver
 * Sent from the region to the data server
 * to note that the region's teleportation landing status has changed
 *
 * <p>Template: {@code TeleportLandingStatusChanged Low 147 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class TeleportLandingStatusChanged : SLMessage() {
    @JvmField var RegionData_Field: RegionData = RegionData()

    /** Block RegionData, Single. */
    open class RegionData {
        @JvmField var RegionHandle: Long = 0L
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 12
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleTeleportLandingStatusChanged(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 147 (TeleportLandingStatusChanged).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x00).toByte())
        byteBuffer.put((0x93).toByte())
        packLong(byteBuffer, RegionData_Field.RegionHandle)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        RegionData_Field.RegionHandle = unpackLong(byteBuffer)
    }
}
